ccmbioinfo/lfs
# CARDS-2431: Unify the CSV and S3 exports; HERACLES-10: File exporter to exclude Holter files from S3 export

To test Heracles:
- build with `mvn clean install -Pdocker`
- generate a compose file with `python3 generate_compose_yaml.py --dev_docker_image  --composum --cards_project cards4heracles --s3_test_container --oak_filesystem`
- start it up with `docker-compose build && docker-compose up`
- create Holter and CPET - External forms, fill them in, upload some files in the File answers, small ones for CPET, big ones (2G) for Holter
- visit `http://localhost:9001/` and create a bucket named `uhn`
- open `http://localhost:8080/Subjects.export?config=Heracles%20-%20Subjects%20with%20Data` and `http://localhost:8080/Subjects.export?config=Heracles%20-%20Big%20Files` to export, wait a bit since transferring many GB of data takes a while
- open `http://localhost:9001/buckets/uhn/browse`
- check that the CPET files are embedded inline in the subject json, but the Holter files aren't
- check that the large Holter files are present, download one and compare against the original to make sure no bytes were lost or mangled
- shut down and cleanup with `docker-compose rm -vf && ./cleanup.sh`

To test YE:

- generate a compose file with `python3 generate_compose_yaml.py --cards_project cards4prems --oak_filesystem --dev_docker_image --composum`
- edit the generated `docker-compose.yml` file and add in `cardsinitial/volumes` `    - ./EXPORTS:/csv-export`
- start it up with `docker-compose build && docker-compose up`
- create a visit, fill it in as a patient and submit it
- open `http://localhost:8080/Questionnaires.export?config=UHN-Survey-Events` to export Survey Events
- on the local disk, open the EXPORTS folder and make sure that the one Survey Events form was not included in the CSV, since it is incomplete
- use composum to add a value to the `Invitation email sent on` answer for the Survey Events form
    - open the form, see its node name in the address bar
    - open composum, open the form
    - find the answer with INCOMPLETE flags (fourth from the bottom)
    - add a property with Type=Date, Name=value, Value=any date
- open `http://localhost:8080/Questionnaires.export?config=UHN-Labeled-Forms`, `http://localhost:8080/Questionnaires.export?config=UHN-Raw-Forms` and `http://localhost:8080/Questionnaires.export?config=UHN-Survey-Events` to export
- on the local disk, open the EXPORTS folder and make sure that all the forms were exported correctly, including the now complete Survey Events
- shut down and cleanup with `docker-compose rm -vf && ./cleanup.sh`
